cluster/cluster.go

```diff
@@ -563,3 +571,116 @@ func MustGetSegmentConfiguration(connection *dbconn.DBConn, getMirrors ...bool)
 	gplog.FatalOnError(err)
 	return segConfigs
 }
+
+/*GetSegmentConfigurationFromFile to parse gpsegconfig_dump or similar file to retrieve segment configuration information.
+gpsegconfig_dump file gets created at $COORDINATOR_DATA_DIR/gpsegconfig_dump by fts process
+The gpsegconfig_dump file follows a structured format, as illustrated in the example below:
+1 -1 p p n u 6000 localhost localhost /data/temp1
+2 0 p p n u 6002 localhost localhost /data/temp2
+3 1 p p n u 6003 localhost localhost /data/temp3
+4 2 p p n u 6004 localhost localhost /data/temp4
+
+Recommended use of the api is to get populate the gpdbArray when database is down.
+if the database is up use GetSegmentConfiguration()/MustGetSegmentConfiguration()  API
+
+input:
+confFile - absolute path of the gpsegconfig_dump or similar structured file
+(e.g. /Users/shrakesh/workspace/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1/gpsegconfig_dump)
+
+return:
+Slice of populated SegConfig objects or else nil
+error if any or else nil.*/
+
+func GetSegmentConfigurationFromFile(confFile string) ([]SegConfig, error) {
```

Comment:
Won't the file always be named "gpsegconfig_dump" ? Why should the utility have to know this filename ? They can just pass the data directory and the function can append "gpsegconfig_dump" to the path
The only use case for asking the utility to pass the name is if this function can be called on arbitrary files that are not named gpsegconfig_dump

Reply:
the initial plan was to just fetch the $COORDINATOR_DATA_DIRECTORY/gpsegconfig_dump file and do the parsing on that, keeping it simple. but later we realized that since this is common go library for gp, we should provide an option to give the file path as an arg for arbitrary files.  this will also enable us to extend the API to parse different type of file format like (json) which we are using in project spine.  Yes we may have to add and enhance the logic but the signature will be same.

Reply:
But this function will specifically be used for parsing the contents of gpsegconfig_dump. Unless we also want to use this function for a file with a different name but similar format/contents, the caller shouldn't really need to know about the existence of `gpsegconfig_dump`

Reply:
Are there any other type of file that can be read that's similar to `gpsegconfig_dump`?
I'm leaning towards `gpsegconfig_dump` as the name, but not a stopper.

Reply:
> Are there any other type of file that can be read that's similar to `gpsegconfig_dump`? I'm leaning towards `gpsegconfig_dump` as the name, but not a stopper.

currently there is no other file similar to gpsegconfig_dump. as now we have decided to pass datadir as arg and appended gpsegconfig to create path, IMO now gpsegconfig_dump variable name look more sensible.